packages/git/src/browser/history/git-history-contribution.ts

```diff
@@ -101,6 +105,14 @@ export class GitHistoryContribution extends AbstractViewContribution<GitHistoryW
         });
     }
 
+    registerKeybindings(keybindings: KeybindingRegistry): void {
```

Comment:
You don't need that. It is done in the AbstractViewContribution.

Reply:
The reason I had to do this was since we define our own command to open the `git-history` with an undefined URI when opening through the menu. Since we define our own command and do not use the one from `AbstractViewContribution` I had to also define it's keybinding. If I remove the keybinding the following occurs:

![git-history](https://user-images.githubusercontent.com/40359487/47495029-25882580-d821-11e8-8e99-3b3dfedbada0.gif)

I assume Theia believes that the `AbstractViewContribution` and our custom open command are different and duplicate the `Git History` command, one without the keybinding present.

I may be wrong however, the patch may be small but  the behaviour was quite tricky to understand at first.

Reply:
That's strange. I really can't confirm this. I just removed the `registerKeybindings` and got this: 
![git-history](https://user-images.githubusercontent.com/28291421/47496870-77986d80-d858-11e8-8602-33919ec85d19.gif)
In our case the `AbstractViewContribution` does not create a command at all. We overwrite `registerCommands` in the `GitHistoryContribution` and don't call `super.registerCommands`. Actually the most necessary change was [this](https://github.com/theia-ide/theia/pull/3272/files#diff-6d484f024681c503b2a799dc4d264573R41)(adding a label to the command).
Now I wonder why have different results.

Reply:
I'll make the changes on my end and test as well, I did notice it was strange as well.

Reply:
I seem to be getting the result I had before by removing the `registerKeybinding`. Since we are not actually using the `AbstractViewContribution` `toggleCommandId` and `toggleKeybinding` and are overwriting with `registerCommands`, should we remove them in favour of using `registerCommands` and `registerKeybindings`?

Reply:
Yes, you're right. That could actually help. :-) Give it a try.

Reply:
It's such an odd case, on Linux I get the two commands, while Mac I have the one. If I remove the `toggleCommandId`, and `toggleKeybinding`, then I don't have the menu item anymore so it would need to be explicitly defined in the `registerMenus`.